pkg/catalog/catalog.go

```diff
@@ -129,8 +131,71 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
 		imageStreamTagMap[imageStreamTag.Name] = imageStreamTag
 	}
 
-	var builderImages []CatalogImage
+	builderImages := getBuildersFromImageStreams(imageStreams, imageStreamTagMap)
+
+	return builderImages, nil
+}
 
+// SliceSupportedTags splits the tags in to fully supported and unsupported tags
+func SliceSupportedTags(catalogImage CatalogImage) ([]string, []string) {
+
+	var supTag, unSupTag []string
+	tagMap := createImageTagMap(catalogImage.imageStreamRef.Spec.Tags)
+	for _, tag := range catalogImage.NonHiddenTags {
+		imageName := tagMap[tag]
+		if isSupportedImage(imageName) {
+			supTag = append(supTag, tag)
+		} else {
+			unSupTag = append(unSupTag, tag)
+		}
+	}
+	return supTag, unSupTag
+}
+
+// createImageTagMap takes a list of image TagReferences and creates a map of type tag name => image name e.g. 1.11 => openshift/nodejs-11
+func createImageTagMap(tagRefs []imagev1.TagReference) map[string]string {
+	tagMap := make(map[string]string)
+	for _, tagRef := range tagRefs {
+		imageName := tagRef.From.Name
+		if tagRef.From.Kind == "DockerImage" {
+			// we get the image name from the repo url e.g. registry.redhat.com/openshift/nodejs:10 will give openshift/nodejs:10
+			urlImageName := strings.SplitN(imageName, "/", 2)[1]
+			// here we remove the tag and digest
+			ns, img, _, _, _ := occlient.ParseImageName(urlImageName)
+			imageName = ns + "/" + img
+		} else if tagRef.From.Kind == "ImageStreamTag" {
+			tagList := strings.Split(imageName, ":")
+			tag := tagList[len(tagList)-1]
+			// if the kind is a image stream tag that means its pointing to an existing dockerImage or image stream image
+			// we just look it up from the tapMap we already have
+			imageName = tagMap[tag]
+		}
+		tagMap[tagRef.Name] = imageName
+	}
+	return tagMap
+}
+
+func isSupportedImage(imgName string) bool {
```

Comment:
these images were figured out by @kadel by going through them manually. this would be the file https://github.com/openshift/odo-init-image/blob/master/language-scripts/image-mappings.json where the mapping exists

Reply:
added the above link as a comment